coderd/healthcheck/provisioner.go

```diff
@@ -0,0 +1,136 @@
+package healthcheck
+
+import (
+	"context"
+	"time"
+
+	"golang.org/x/mod/semver"
+
+	"github.com/coder/coder/v2/buildinfo"
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/db2sdk"
+	"github.com/coder/coder/v2/coderd/database/dbauthz"
+	"github.com/coder/coder/v2/coderd/database/dbtime"
+	"github.com/coder/coder/v2/coderd/healthcheck/health"
+	"github.com/coder/coder/v2/coderd/provisionerdserver"
+	"github.com/coder/coder/v2/coderd/util/apiversion"
+	"github.com/coder/coder/v2/coderd/util/ptr"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/provisionersdk"
+)
+
+// @typescript-generate ProvisionerDaemonsReport
+type ProvisionerDaemonsReport struct {
+	Severity  health.Severity  `json:"severity"`
+	Warnings  []health.Message `json:"warnings"`
+	Dismissed bool             `json:"dismissed"`
+	Error     *string          `json:"error"`
+
+	ProvisionerDaemons []codersdk.ProvisionerDaemon `json:"provisioner_daemons"`
+}
+
+type ProvisionerDaemonsReportDeps struct {
+	// Required
+	CurrentVersion         string
+	CurrentAPIMajorVersion int
+	Store                  ProvisionerDaemonsStore
+
+	// Optional
+	TimeNowFn     func() time.Time // Defaults to dbtime.Now
```

Comment:
This got lost in a resolved comment, but I feel `Fn` suffix can be dropped as it's part of the type and becomes redundant.